tests/api_test.py

```diff
@@ -2068,6 +2068,27 @@ def test_linearize_aval_error(self):
     with self.assertRaisesRegex(ValueError, "tangent values inconsistent"):
       f_jvp(np.ones(2, np.int32))
 
+  def test_dunder_jax_array(self):
+    # https://github.com/google/jax/pull/4725
+
+    class AlexArray:
+      def __init__(self, jax_val):
+        self.jax_val = jax_val
+      def __jax_array__(self):
+        return self.jax_val
+      dtype = property(lambda self: self.jax_val.dtype)
+      shape = property(lambda self: self.jax_val.shape)
+
+    x = AlexArray(jnp.array([1., 2., 3.]))
+    y = jnp.sin(x)
```

Comment:
It's a bit strange that this returns a jax array rather than wrapping the output in an AlexArray. I wonder if it wouldn't be better to provide something analogous to [NEP 18](https://numpy.org/neps/nep-0018-array-function-protocol.html) or some of the followups to that?